includes/RemovePIIJob.php

```diff
@@ -0,0 +1,482 @@
+<?php
+
+namespace Miraheze\RemovePII;
+
+use CentralAuthUser;
+use Exception;
+use ExtensionRegistry;
+use GenericParameterJob;
+use Job;
+use MediaWiki\MediaWikiServices;
+use Title;
+use User;
+use UserProfilePage;
+use WikiPage;
+
+class RemovePIIJob extends Job implements GenericParameterJob {
+	/** @var string */
+	private $database;
+
+	/** @var string */
+	private $oldName;
+
+	/** @var string */
+	private $newName;
+
+	/**
+	 * @param array $params
+	 */
+	public function __construct( array $params ) {
+		parent::__construct( 'RemovePIIJob', $params );
+
+		$this->database = $params['database'];
+		$this->oldName = $params['oldname'];
+		$this->newName = $params['newname'];
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function run() {
+		$newCentral = CentralAuthUser::getInstanceByName( $this->newName );
+
+		// Invalidate cache before we begin transaction
+		$newCentral->invalidateCache();
+
+		$oldName = User::newFromName( $this->oldName );
+		$newName = User::newFromName( $this->newName );
+
+		$userOldName = $oldName->getName();
+		$userNewName = $newName->getName();
+
+		if ( !$newName ) {
+			$this->setLastError( "User {$userNewName} is not a valid name" );
+
+			return false;
+		}
+
+		$userId = $newName->getId();
+
+		if ( !$userId ) {
+			$this->setLastError( "User {$userNewName} ID equal to 0" );
+
+			return false;
+		}
+
+		$dbw = wfGetDB( DB_MASTER, [], $this->database );
+
+		$userActorId = $newName->getActorId( $dbw );
+
+		$tableUpdates = [
```

Comment:
Shouldn't this be configured in LocalSettings.php and/or through extension hooks? Not only to support third-party usage, but to avoid making lots of commits to this extension as well.

Reply:
Yes I had planned to add hook support at least and make this a configuration, but was planning to split that into another PR. But can do that now if you think I should.